test/language/statements/class/elements/private-field-is-visible-in-computed-properties.js

```diff
@@ -10,20 +10,46 @@ info: |
     2. Let classScope be NewDeclarativeEnvironment(lex).
     3. Let classScopeEnvRec be classScope's EnvironmentRecord.
     ...
+    8. If ClassBodyopt is present, then
+        a. For each element dn of the PrivateBoundIdentifiers of ClassBodyopt,
+          i. Perform classPrivateEnvRec.CreateImmutableBinding(dn, true).
+    ...
     15. Set the running execution context's LexicalEnvironment to classScope.
     16. Set the running execution context's PrivateEnvironment to classPrivateEnvironment.
     ...
     27. For each ClassElement e in order from elements
       a. If IsStatic of e is false, then
         i. Let field be the result of ClassElementEvaluation for e with arguments proto and false.
     ...
+
+  MemberExpression : MemberExpression . PrivateIdentifier
+    ...
+    5. Return MakePrivateReference(bv, fieldNameString).
+
+  MakePrivateReference ( baseValue, privateIdentifier )
+    ...
+    2. Let privateNameBinding be ? ResolveBinding(privateIdentifier, env).
+    3. Let privateName be GetValue(privateNameBinding).
+    ...
+
+  GetValue (V)
+    ...
+    6. Else,
+      a. Assert: base is an Environment Record.
+      b. Return ? base.GetBindingValue(GetReferencedName(V), IsStrictReference(V)).
+
+  GetBindingValue ( N, S )
+    ...
+    3. If the binding for N in envRec is an uninitialized binding, throw a ReferenceError exception.
+    ...
+
 features: [class-fields-private, class]
 ---*/
 
-assert.throws(TypeError, function() {
+assert.throws(ReferenceError, function() {
   class C {
     [this.#f] = 'Test262';
```

Comment:
The usage of `this` here makes this test wrong into strict mode (see discussion in https://github.com/tc39/test262/pull/2241#issuecomment-520872478). We should use any other base (for example using `self` to properly catch global object) or flag this test as `noStrict`. I prefer the former approach.

Reply:
Thanks for catching that, fixed